File: Metadata-migration/models/metadata_asset.py
from dataclasses import dataclass, field
from typing import Optional, List


@dataclass
class MetadataAsset:
    uid: str
    type: str
    name: str
    description: Optional[str]      
    owner: Optional[str]         
    created_on: Optional[str]        
    updated_on: Optional[str]        
    tags: List[str] = field(default_factory=list)  # Always List, never None

File: MetadataApi/Models/MetadataAsset.cs
using System.Collections.Generic;

namespace MetadataApi.Models
{
    /// <summary>
    /// Represents the internal domain model for a metadata asset.
    /// This model is used within the service layer and is never exposed directly via the API.
    /// </summary>
    public class MetadataAsset
    {
        public string Uid { get; set; } = string.Empty;
        public string Type { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? Owner { get; set; }
        public string SourceSystem { get; set; } = string.Empty;
        public List<string> Tags { get; set; } = new();
    }
}
